classes/output/renderer.php

```diff
@@ -1385,7 +1385,11 @@ private function construct_user_sessions_log(user_data $userdata) {
                         // URL to the page that lets the student modify their attendance.
                         $url = new moodle_url('/mod/attendance/attendance.php',
                                 array('sessid' => $sess->id, 'sesskey' => sesskey()));
-                        $cell = new html_table_cell(html_writer::link($url, get_string('submitattendance', 'attendance')));
+                        if (time() < $sess->sessdate) {
```

Comment:
I feel like we might need an api function "is open for marking" - because some sessions will be open for marking "early" and so still need to show the normal string and it's a bit messy how the rest of the code does this inconsistently.

Reply:
I have fixed this and did a force push to add it in the same commit. Let me know if in the future you prefer the feedback work to go in a separate commit.